content/docs/gke/pipelines/authentication-pipelines.md

```diff
@@ -9,49 +9,117 @@ This page describes authentication for Kubeflow Pipelines to GCP.
+#### Authoring pipelines to use Workload Identity
+
+Pipelines don't need any specific changes to authenticate to GCP, it will use the GSA bound to the KSA it uses transparently.
+
+However, existing pipelines that use [use_gcp_secret kfp sdk operator](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.extensions.html#kfp.gcp.use_gcp_secret) need to remove the `use_gcp_secret` usage to use the bound GSA.
+You can also continue to use `use_gcp_secret` in a cluster with Workload Identity enabled, but pipeline steps with `use_gcp_secret` will use the GSA corresponding to the secret provided.
+
+#### Cluster setup to use Workload Identity for Pipelines Standalone or GCP Hosted ML pipelines
+
+##### 1. Create your cluster with Workload Identity enabled
+
+* In Google Cloud Console UI, you can enable it in `Create a Kubernetes cluster -> Security -> Enable Workload Identity` like the following:
+<img src="/docs/images/pipelines/gke-enable-workload-identity.png">
+
+* Using `gcloud` CLI, you can enable it with:
+```bash
+gcloud beta container clusters create cluster-name \
+  --release-channel regular \
+  --workload-pool=project-id.svc.id.goog
+```
 
-Pipelines don't need any change to authenticate to GCP, it will use the GSA bound to `pipeline-runner` KSA transparently.
+References:
 
-However, existing pipelines that use [use_gcp_secret kfp sdk operator](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.extensions.html#kfp.gcp.use_gcp_secret) need to remove `use_gcp_secret` usage to use the bound GSA.
-You can also continue to use `use_gcp_secret` in a cluster with workload identity enabled, but pipeline steps with `use_gcp_secret` will use the GSA corresponding to the secret provided.
+* [Enable Workload Identity on a new cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_workload_identity_on_a_new_cluster)
 
-#### Cluster setup to use workload identity for Pipelines Standalone
+* [Enable Workload Identity on an existing cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_workload_identity_on_an_existing_cluster)
 
-**After you enabled workload identity**, you need to bind workload identities for KSAs used by [Pipelines Standalone](/docs/pipelines/installation/overview/#kubeflow-pipelines-standalone).
-The following helper bash scripts bind workload identities for KSAs provided by Pipelines Standalone:
+##### 2. Deploy Kubeflow Pipelines
+Deploy via [Pipelines Standalone](/docs/pipelines/installation/overview/#kubeflow-pipelines-standalone) or [GCP hosted ML pipelines](/docs/pipelines/installation/overview#gcp-hosted-ml-pipelines) as usual.
+
+Note, for GCP hosted ML pipelines additionally, you need to patch `proxy-agent` to work with workload identity:
+```
+kubectl patch deployment proxy-agent --patch '{"spec": {"template": {"spec": {"hostNetwork": true}}}}'
+```
+
+##### 3. Bind Workload Identities for KSAs used by Kubeflow Pipelines
```

Comment:
Thanks! That's a good suggestion. Let me add one

Reply:
PTAL

Reply:
I would recommend demonstrating the recommended way to use the command. For example, it seems like the following steps may be useful to someone:

1. Create a GSA
1. Add roles to the GSA (maybe read/write access to GCS since that seems like it would be a common need)
1. Bind the GSA to the correct KSA.

Reply:
`gcp-workload-identity-setup.sh` is a no configuration (only configurations for basic info, but creating gsa and binding gsa are automatic) setup script that does all of 1. and 3.

User is required to do 2. separately.

Reply:
Ok, if the script is meant to be the documentation then remove the example. If you think that the general usage will be stable (call the command to create a GSA and bind it to a KSA) then consider demonstrating how to do that.

Reply:
@joeliedtke I think current example is helpful for recommending the default path of using `gcp-workload-identity-setup.sh`.